src/utilities/loading-utils.ts

```diff
@@ -0,0 +1,35 @@
+import { DEBUG_LOADING } from "../lib/debug";
+
+const sessionStorageStartTimeItem = 'loading-time-start';
+const sessionStorageMessageItem = 'loading-message';
+const messageSeparator = '<br/>';
+
+function getTimestamp() {
+  const start = sessionStorage.getItem(sessionStorageStartTimeItem);
+  if (!start) {
+    return "??";
+  }
+  return Math.floor(performance.now()-Number.parseFloat(start));
+}
+
+export function getCurrentLoadingMessage() {
+  return sessionStorage.getItem(sessionStorageMessageItem) || '...';
+}
+
+export function showLoadingMessage(msg: string) {
+  if (DEBUG_LOADING) {
+    console.log(`Loading @${getTimestamp()}ms: ${msg}`);
```

Comment:
This line and the line below should have a lint comment added to them to allow console statements. The dev lint rules allow console.log statements but the production rules do not. You can see these warnings locally by running `npm run eslint:build`.